pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/TopicsImpl.java

```diff
@@ -1397,9 +1399,14 @@ public OffloadProcessStatus offloadStatus(String topic)
         WebTarget path = topicPath(tn, "offload");
         final CompletableFuture<OffloadProcessStatus> future = new CompletableFuture<>();
         asyncGetRequest(path,
-                new InvocationCallback<OffloadProcessStatus>() {
+                new InvocationCallback<String>() {
                     @Override
-                    public void completed(OffloadProcessStatus offloadProcessStatus) {
+                    public void completed(String jsonString) {
+                        Gson gson = new GsonBuilder().registerTypeAdapter(MessageId.class,
```

Comment:
We should use Jackson for all the JSON ser/de, unless there's a particular reason not to do so. Please use `ObjectMapperFactory.getThreadLocal()....`

Reply:
https://github.com/freeznet/pulsar/commit/fa05e389b99f2d6aa90f4775f487f281b38796dc
here is a sample commit that changes `Gson` to `Jackson`.

Reply:
We want to make sure to use Jackson, through the `ObjectMapperFactory.getThreadLocal()` since it comes configured with some settings. 

If `@JsonDeserialize` annotations are the problem, there will be bigger problems porting the POJOs defined in `pulsar-common`.

The better option here is to untie the `pulsar-client-admin-api` from the POJOs. Instead, `pulsar-client-admin-api` should return Java interfaces instead of POJOs that are used for serialization.

Reply:
@merlimat Indeed, there will be some problems with `pulsar-common`, so how about we use `Map` or `List` as the return type in `pulsar-client-admin-api`? I havnt tried it out yet, any suggestions? thanks again.